### ISSUE TITLE
File extension checks need to take into account case-sensitive file systems

### ISSUE THREAD

**Issue by SteveL-MSFT** [PowerShell/PowerShell#3218]
https://github.com/PowerShell/PowerShell/search?utf8=%E2%9C%93&q=ordinalignorecase+getextension&type=Code

(query may not be complete)

On Linux/Mac, we should follow any precedent dotnet has already defined for assemblies (in case there's foo.dll and foo.DLL)

**Comment by iSazonov** [comment]
I believe we should consider not only extensions but paths.

**Comment by SteveL-MSFT** [comment]
Need to take in account that fat32 mounted on Linux recognizes the case-insensitiveness

**Comment by iSazonov** [comment]
@SteveL-MSFT Linux can mount NTFS too but I don't about case-insensitiveness.

**Comment by jeffbi** [comment]
@iSazonov  From https://support.microsoft.com/en-us/help/100108/overview-of-fat,-hpfs,-and-ntfs-file-systems :

> Names preserve case, but are not case sensitive. NTFS makes no distinction of filenames based on case.

**Comment by joeyaiello** [comment]
@jeffbi: is this fixed by #3441? Or do we just need to take the same system-call based approach throughout more of the codebase?

**Comment by jeffbi** [comment]
@joeyaiello No, the code in #3441 is currently used only by the Copy-Item cmdlet.
